src/main/scala/com/fulcrumgenomics/util/UpdateGffContigNames.scala

```diff
@@ -0,0 +1,151 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2019 Fulcrum Genomics LLC
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.fulcrumgenomics.util
+
+import java.io.BufferedWriter
+
+import com.fulcrumgenomics.FgBioDef._
+import com.fulcrumgenomics.cmdline.{ClpGroups, FgBioTool}
+import com.fulcrumgenomics.commons.collection.BetterBufferedIterator
+import com.fulcrumgenomics.commons.util.{LazyLogging, StringUtil}
+import com.fulcrumgenomics.sopt.{arg, clp}
+
+
+@clp(description =
+  """
+    |Updates then contig names in a GFF.
+    |
+    |The first column of the input is the source name and the second column is an target name.  If there
+    |is more than one target names (ex. multiple alternates), each alternate name may be on a separate line or as
+    |additional columns.  Only the first target name will be considered.  This is mainly to support the output of
+    |`CollectAlternateContigNames`.
+    |
+    |Please note: the output GFF will be in the same order as the input GFF.
+  """,
+  group = ClpGroups.Utilities)
+class UpdateGffContigNames
+(@arg(flag='i', doc="Input GFF.") val input: FilePath,
+ @arg(flag='m', doc="The path to the source to target names.") val mapping :FilePath,
+ @arg(flag='o', doc="Output GFF.")val output: FilePath,
+ @arg(doc="Skip missing contigs.") val skipMissing: Boolean = false
+) extends FgBioTool with LazyLogging {
+
+  Io.assertReadable(input)
+  Io.assertReadable(Seq(input, mapping))
+  Io.assertCanWriteFile(output)
+
+  private val sequenceRegionKey = "##sequence-region"
+
+
+  override def execute(): Unit = {
+    val progress = ProgressLogger(logger, noun="features", verb="written", unit=10e5.toInt)
+
+    val srcToTarget = Io.readLines(mapping).map { line =>
+      val fields = line.split('\t')
+      require(fields.length <= 2, s"Malformed line: expected at least two columns: $line")
+      (fields(0), fields(1))
+    }.toMap
```

Comment:
This validation is similar to to UpdateFastaContigNames.scala:70 and I think these inputs are the same. The only difference being that one adds to "seen" set. Seems like it could be refactored though.

Reply:
I've added a 'ContigNameMapping' class to parse the mapping file and handle many to one src-> target mappings.